crates/server/src/server/server_future.rs

```diff
@@ -48,25 +53,27 @@ impl<T: RequestHandler> ServerFuture<T> {
         Self {
             handler: Arc::new(handler),
             join_set: JoinSet::new(),
+            drain: Drain::new(),
         }
     }
 
     /// Register a UDP socket. Should be bound before calling this function.
-    pub fn register_socket(&mut self, socket: net::UdpSocket) {
+    pub fn register_socket(&mut self, socket: tokio::net::UdpSocket) {
         debug!("registering udp: {:?}", socket);
 
         // create the new UdpStream, the IP address isn't relevant, and ideally goes essentially no where.
         //   the address used is acquired from the inbound queries
-        let (mut buf_stream, stream_handle) =
+        let (stream, stream_handle) =
             UdpStream::with_bound(socket, ([127, 255, 255, 254], 0).into());
-        //let request_stream = RequestStream::new(buf_stream, stream_handle);
+        let mut stream = stream.take_until(self.drain.clone());
```

Comment:
Why do we need this change?

Reply:
This just adds a line that wraps the original stream with one that will drain as we need.